shell/platform/darwin/ios/framework/Source/FlutterDartVMServicePublisher.mm

```diff
@@ -83,18 +88,20 @@ - (void)publishServiceProtocolPort:(NSURL*)url {
   // Physical devices need to request all interfaces.
   uint32_t interfaceIndex = 0;
 #endif  // TARGET_IPHONE_SIMULATOR
-  const char* registrationType = "_dartobservatory._tcp";
+  const char* registrationType = "_dartVmService._tcp";
+  const char* legacyRegistrationType = "_dartobservatory._tcp";
+
   const char* domain = "local.";  // default domain
   uint16_t port = [[url port] unsignedShortValue];
 
-  NSData* txtData = [FlutterObservatoryPublisher createTxtData:url];
+  NSData* txtData = [FlutterDartVMServicePublisher createTxtData:url];
   int err = DNSServiceRegister(&_dnsServiceRef, flags, interfaceIndex,
-                               FlutterObservatoryPublisher.serviceName.UTF8String, registrationType,
-                               domain, NULL, htons(port), txtData.length, txtData.bytes,
-                               RegistrationCallback, NULL);
+                               FlutterDartVMServicePublisher.serviceName.UTF8String,
+                               registrationType, domain, NULL, htons(port), txtData.length,
+                               txtData.bytes, RegistrationCallback, NULL);
 
   if (err != 0) {
-    FML_LOG(ERROR) << "Failed to register observatory port with mDNS with error " << err << ".";
+    FML_LOG(ERROR) << "Failed to register Dart VM Service port with mDNS with error " << err << ".";
```

Comment:
Networking in iOS is a protected resource and therefore requires permissions, specifically NSBonjourServices. Without it, this will always error and show this error message to users.

In the Flutter tools, we add in NSBonjourServices to the Info.plist on build here: 
https://github.com/flutter/flutter/blob/c31856bc48596d251ade554c33a09184e69185d9/packages/flutter_tools/bin/xcode_backend.dart#L247-L285

To avoid errors, the tool should be updated first to include the new `_dartVmService._tcp` (and also keep the original `_dartobservatory._tcp`).

Reply:
This code is falling back to `_dartobservatory._tcp` with the `legacyRegistrationType` check below so I think it should still work. However, it shouldn't log any errors like this of or the iOS 14 one until both fail.

@vashworth's suggestion to add both to the tool first would also work.

Reply:
Updated to only output the error message after both registration attempts have failed.